crates/foundry/forge/src/lib.rs

```diff
@@ -38,19 +46,46 @@ impl TestOptions {
     /// Returns a "fuzz" test runner instance.
     pub fn fuzz_runner(&self) -> TestRunner {
         let fuzz_config = self.fuzz_config().clone();
-        let failure_persist_path = fuzz_config
-            .failure_persist_dir
-            .unwrap()
-            .join(fuzz_config.failure_persist_file.unwrap())
-            .into_os_string()
-            .into_string()
-            .unwrap();
-        self.fuzzer_with_cases(
-            fuzz_config.runs,
-            Some(Box::new(FileFailurePersistence::Direct(
-                failure_persist_path.leak(),
-            ))),
-        )
+
+        if let Some(failure_persist_dir) = fuzz_config.failure_persist_dir {
+            let failure_persist_path = failure_persist_dir
+                .join(fuzz_config.failure_persist_file)
+                .into_os_string()
+                .into_string()
+                .expect("path should be valid UTF-8");
+
+            // HACK: We need to leak the path as
```

Comment:
For correctness, I believe we should still drop the leaked strings upon static deinitialization. That can be achieved by adding a local `struct SomeName { inner: HashSet<&'static str> }` which has a `impl Drop`. This destructor would be called when the `static FAILURE_PATHS` is dropped, guaranteeing that our application doesn't report any memory leaks in tooling.

Reply:
Hey, I tried this out with LLVM leak sanitizer and it reports errors in the base branch, but not in this one, so I think it's ok as is.

Test command (tried on ARM linux in Docker): `RUSTFLAGS="-Z sanitizer=leak" cargo +nightly test -p forge test_fuzz`

Reply:
(I couldn't get Miri working on the `forge` integration tests)

Reply:
Thanks for validating with leak sanitizer.

[This](https://stackoverflow.com/a/71419608/4944080) is the reason why Leak Sanitizer would not be able to detect a "memory leak" of this kind. It depends on your definition of memory leak, though. Leak sanitizer doesn't consider it a leak because a static variable is still referring to the address of the originally allocated strings.